cmd/clusterctl/client/move.go

```diff
@@ -146,25 +149,32 @@ func (c *clusterctlClient) Backup(options BackupOptions) error {
 }
 
 func (c *clusterctlClient) Restore(options RestoreOptions) error {
-	// Get the client for interacting with the source management cluster.
-	toCluster, err := c.clusterClientFactory(ClusterClientFactoryInput{Kubeconfig: options.ToKubeconfig})
+	toCluster, err := c.getClusterClient(options.ToKubeconfig)
 	if err != nil {
 		return err
 	}
 
-	// Ensure this command only runs against management clusters with the current Cluster API contract.
-	if err := toCluster.ProviderInventory().CheckCAPIContract(); err != nil {
+	if _, err := os.Stat(options.Directory); os.IsNotExist(err) {
 		return err
 	}
 
-	// Ensures the custom resource definitions required by clusterctl are in place.
-	if err := toCluster.ProviderInventory().EnsureCustomResourceDefinitions(); err != nil {
-		return err
+	return toCluster.ObjectMover().Restore(toCluster, options.Directory)
```

Comment:
Also `ObjectMover().Restore` and `ObjectMover().Backup` should be deprecated in favour of `ObjectMover().FromDirectory` and `ObjectMover().ToDirectory`